pkg/cluster/internal/providers/provider/common/logs.go

```diff
@@ -0,0 +1,60 @@
+package common
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+	"sigs.k8s.io/kind/pkg/errors"
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// CollectLogs provides the common functionality
+// to get various debug info from the node
+func CollectLogs(n nodes.Node, dir string) error {
+	execToPathFn := func(cmd exec.Cmd, path string) func() error {
+		return func() error {
+			f, err := FileOnHost(filepath.Join(dir, path))
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			return ExecToWriter(cmd, f)
+		}
+	}
+	return errors.AggregateConcurrent([]func() error{
+		// record info about the node container
+		execToPathFn(
+			n.Command("cat", "/kind/version"),
+			"kubernetes-version.txt",
+		),
+		execToPathFn(
+			n.Command("journalctl", "--no-pager"),
+			"journal.log",
+		),
+		execToPathFn(
+			n.Command("journalctl", "--no-pager", "-u", "kubelet.service"),
+			"kubelet.log",
+		),
+		execToPathFn(
+			n.Command("journalctl", "--no-pager", "-u", "containerd.service"),
+			"containerd.log",
+		),
+	})
+}
+
+// ExecToWriter is a helper to run a cmd and write the output to writer
+func ExecToWriter(cmd exec.Cmd, w io.Writer) error {
+	cmd.SetStdout(w)
+	cmd.SetStderr(w)
+	return cmd.Run()
```

Comment:
this can be done inline in other code like `cmd.SetStdout(w).SetStderr(w).Run()`, a one liner like this isn't usually an exported method /shrug

Reply:
also technically per the cmd interface this version should be:
```
...
cmd = cmd.SetStdout(w)
cmd = cmd.SetStderr(w)
...
```

Reply:
yeah I had that previously, guess not really required as a helper.